### ISSUE TITLE
Polishing of CDI Utilities 1

### ISSUE THREAD

**Issue by adriansteffan** [langcog/peekbankr#19]
I have added a first version of utilities to deal with cdi data, but there are some areas where they could use polishing:

A) Validation of CDI data 

- [ ] there now exists the function cleanup_cdi_data that is supposed to catch anomalies that make analysis difficult. For now, it only catches duplicate cdi values for the same administration (we need a validator for this anyway - see https://github.com/peekbank/peekbank-data-import/issues/146) - we should brainstorm and implement other anomalies that we would like to catch here

B) Relative CDI scores
- [ ] I took some code from the peekbank method repo that calculates a percentage of points reached out of the maximum for a given cdi type. However, I could use the eyes of someone experienced with cdi to determine the correct comparison values for various languages and cdi types.
 

C) Percentile calculation for CDI rawscores (benchmarking)

- [ ] I have added a function that gets the respective cdi percentile for each rawscore loosely based on work done by george and the others https://github.com/kachergis/cdi-percentiles/tree/main. At the moment, this is hardcoded for American English and the 2022 norm, so we might want to add more languages and norms and make the function more flexible as a result. Additionally, this function could use some testing + a look by someone more experienced with cdis (open point: what to do with kids that are too young or old? right now, the function defaults to the nearest available age)

**Comment by mzettersten** [comment]
@mcfrank Adrian has been working on CDI score handling, including computing CDI percentiles. Looping you in here!

**Comment by mzettersten** [comment]
D) CDI short form percentiles for English

- [ ] We still need to find/ add CDI percentiles for the English shortform in particular (also other languages, but these are less relevant to immediate analyses).

E) Fix bug creating NAs for zero scores

- [ ] something's breaking the code for zero scores

**Comment by mcfrank** [comment]
@kachergis this seems like you might have thoughts about A and B? 
@vmarchman do you have percentile tables for the English short form?

**Comment by vmarchman** [comment]
Yes - Here you go.  Not sure if you wanted both or boys/girls separately,
so here's both.  Level I = WG, Level IIA/B = two forms of WS.

**Comment by vmarchman** [comment]
[LevelI_comp_both_weighted.csv](https://github.com/user-attachments/files/18340609/LevelI_comp_both_weighted.csv)
[LevelI_comp_boys_weighted.csv](https://github.com/user-attachments/files/18340610/LevelI_comp_boys_weighted.csv)
[LevelI_comp_girls_weighted.csv](https://github.com/user-attachments/files/18340611/LevelI_comp_girls_weighted.csv)
[LevelI_prod_both_weighted.csv](https://github.com/user-attachments/files/18340612/LevelI_prod_both_weighted.csv)
[LevelI_prod_boys_weighted.csv](https://github.com/user-attachments/files/18340613/LevelI_prod_boys_weighted.csv)
[LevelI_prod_girls_weighted.csv](https://github.com/user-attachments/files/18340614/LevelI_prod_girls_weighted.csv)
[LevelIIA_prod_boys_weighted.csv](https://github.com/user-attachments/files/18340615/LevelIIA_prod_boys_weighted.csv)
[LevelIIA_prod_both_weighted.csv](https://github.com/user-attachments/files/18340616/LevelIIA_prod_both_weighted.csv)
[LevelIIA_prod_girls_weighted.csv](https://github.com/user-attachments/files/18340617/LevelIIA_prod_girls_weighted.csv)
[LevelIIB_boys_weighted.csv](https://github.com/user-attachments/files/18340618/LevelIIB_boys_weighted.csv)
[LevelIIB_girls_weighted.csv](https://github.com/user-attachments/files/18340619/LevelIIB_girls_weighted.csv)
[LevelIIB_prod_both_weighted.csv](https://github.com/user-attachments/files/18340620/LevelIIB_prod_both_weighted.csv)

**Comment by vmarchman** [comment]
Sorry - yes, attached in github now.

-vm
